server/src/main/java/org/elasticsearch/rest/BaseRestHandler.java

```diff
@@ -77,6 +78,13 @@ public final long getUsageCount() {
 
     @Override
     public final void handleRequest(RestRequest request, RestChannel channel, NodeClient client) throws Exception {
+        // check if the query has any parameters that are not in the supported set (if declared)
+        Set<String> supported = supportedQueryParameters();
+        if (supported != null && supported.containsAll(request.params().keySet()) == false) {
```

Comment:
This check seems like it removes the need for the unconsumed parameters infra below, since it more directly checks whether a parameter is supported, rather the current "anything unused must be unsupported"? This can be a followup, but it seems like that could be removed after this change (or at least once all apis have supportedQueryParams setup?)

Reply:
They've got slightly different use cases - see https://github.com/elastic/elasticsearch/pull/106820#discussion_r1562344668

Reply:
I don't think they are different use cases. The reason we check if params are "consumed" was as a way to check if params are valid without explicitly defining which params are valid. But we can discuss if/how to remove param consumption check after merging this one.